src/legacy/ui/public/kfetch/kfetch.ts

```diff
@@ -61,11 +61,17 @@ export async function kfetch(
       });
 
       return window.fetch(fullUrl, restOptions).then(async res => {
-        const body = await getBodyAsJson(res);
-        if (res.ok) {
-          return body;
+        if (!res.ok) {
+          throw new KFetchError(res, await getBodyAsJson(res));
         }
-        throw new KFetchError(res, body);
+        const contentType = res.headers.get('content-type');
+        if (contentType && contentType.split(';')[0] === 'application/json') {
+          return await getBodyAsJson(res);
+        }
+        if (contentType && contentType.split(';')[0] === 'application/ndjson') {
+          return await getBodyAsBlob(res);
+        }
+        return res;
```

Comment:
In order to prevent any bugs, what do you think about changing this to default to `getBodyAsJson` if the content type is not ndjson?